camviewer_ui_impl.py

```
@@ -2098,6 +2060,34 @@ def connectCamera(self, sCameraPv, index, sNotifyPv=None):
                 self.ui.label_status.setText("Zero pixels in image")
                 print("Zero pixels in image (no width/height)")
             return
+        else:
+            self.count = self.rowPv.value * self.colPv.value
+            if self.isColor:
+                self.count *= 3
+
+        if self.lFlags[index] != "":
+            self.bits = int(self.lFlags[index])
+        else:
+            self.bits_pv = self.connectPv(self.cameraBase + ":BitsPerPixel_RBV")
```

self.bits is set to 10 in init. What would be the negative consequences of using ten if we don't get the real number of bits from the camviewer.cfg or bits_pv instead of just failing to connect?

---

This is pretty low stakes I actually, I thought it was used in more places but this is only used to set the bounds for the color mapping. The image processing internals always allocate 32 bits regardless.

If `self.bits` is too big, we'll have extra space in the color map sliders, which isn't so bad.
If `self.bits` is too small, we won't be able to see the full range of the bit depth in the image.

I think it's a judgement call whether or not to allow us to proceed without this PV. A missing PV here is likely an IOC or configuration issue, but the consequence is pretty minor. Maybe I should increase the default from 10 to 12 to match some of these newer gige models and allow it to fail and fall back to 12.

---

One more related issue, the camviewer as written doesn't ever update this. So, if you reconfigure your camera to have a higher bit depth, you can't see those in camviewer unless you re-initialize the camera...

---

Maybe set it to 12 as a default, and let it fail with a console message? That way a user would know about the problem without restricting their camviewer usage completely. 

Updating the record should probably be an issue on ioc-common-gigecam. I guess we would also need a callback to update the range limits if the pv ever updates too, but that's in the future if ever.

---

I think the gigecam IOC already allows you to live update the bit depth, so it's mostly on this side we could add a callback in the future